### DISCUSSION TITLE
Lesson 12: Math operation with PRECISION in "_healthFactor" function

### DISCUSSION THREAD

**Discussion by theooldlyon** [Cyfrin/foundry-full-course-cu#985] [Q&A]
Hi guys, could somebody help me understand how the calculation of the **Health Factor** happens in the **DSCEngine** contract?

I am confused, since **collateralAdjustedForThreshold** is multiplied for **1e18**, will the value returned at the end always be bigger than **1**?

I've done a calculation with:
- **190 $**  deposited in collateral
- **100 $** tokens minted

The health factor is  **0.95 **, which theoretically should authorize liquidation, but since in the contract's logic it's multiplied by **1e18** is much bigger than **1**.

I'm confused 😅

Can anyone help me understand this, please?

```solidity  
function _healthFactor(address user) private view returns (uint256) {
        // total DSC minted
        // total collateral VALUE
        (uint256 totalDscMinted, uint256 collateralValueInUsd) = _getAccountInformation(user);
        uint256 collateralAdjustedForThreshold = (collateralValueInUsd * LIQUIDATION_THRESHOLD) / LIQUIDATION_PRECISION;

        // $150 ETH / 100 DSC = 1.5
        // 150 * 50 = 7500 / 100 = (75 / 100) < 1

        // $1000 ETH / 100 DSC
        // 1000 * 50 = 50000 / 100 = (500 / 100) >1
        return (collateralAdjustedForThreshold * PRECISION) / totalDscMinted;
    }```

**Comment by theirrationalone** [comment]
Maybe [this reply](https://github.com/Cyfrin/foundry-full-course-f23/discussions/523#discussioncomment-6630649) can help you ;`)

**Comment by sunnyStefi** [comment]
@theooldlyon In solidity we have to use **fixed point arithmetic**, which let us represent values with high precision. This is due to solidity which does not support floating point precision. The health factor of the user must always be expressed in e18 notation. In this way **we can represent 0.95 as 95e16.**
